File: HealthAndFitnessPlatformJava/src/main/java/com/example/HealthAndFitnessPlatform/model/WorkoutPlan.java
package com.example.HealthAndFitnessPlatform.model;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Formula;

import java.time.LocalDateTime;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class WorkoutPlan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Enumerated(EnumType.STRING)
    @Column(name = "workout_type", length = 50)
    private WorkoutType workoutType;

    private int duration;

    private double burningCalories;

    @Lob
    private String workoutMessage;

    @ManyToOne
    @JoinColumn(name = "userId")
    private User user;

    @Column(name = "createdAt")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt = LocalDateTime.now();

    @PrePersist
    @PreUpdate
    public void calculateBurningCalories() {
        this.burningCalories = duration * workoutType.getCaloriesBurnedPerMinute();
    }

}

File: HealthAndFitnessPLatformNG/src/app/model/workoutPlan.ts
export class workoutPlan{

          id:number;
          workoutType:string;
          duration:number;
          burningCalories:number;
          workoutMessage:string;
          userId:number;
          createdAt : string;

          constructor(id:number,workoutType:string,duration:number,burningCalories:number,workoutMessage:string,userId:number,createdAt:string){
                         this.id = id;
                         this.workoutType = workoutType;
                         this.duration = duration;
                         this.burningCalories = burningCalories;
                         this.workoutMessage = workoutMessage;
                         this.userId = userId;
                         this.createdAt = createdAt;
          }

}